### PR TITLE
Use timber instead of printstacktrace

### PR DESCRIPTION
**Pull request by abelgardep**
Continuation of #2770 suggested in https://github.com/owncloud/android-library/pull/293#issuecomment-574991808
Needs https://github.com/owncloud/android-library/pull/297

### CODE REVIEW

File: owncloudApp/src/main/java/com/owncloud/android/authentication/AccountAuthenticator.java [hunk 1]
```diff
@@ -97,8 +97,7 @@ public Bundle addAccount(AccountAuthenticatorResponse response,
             try {
                 validateAccountType(accountType);
             } catch (AuthenticatorException e) {
-                Timber.e(e, "Failed to validate account type " + accountType + ": " + e.getMessage());
-                e.printStackTrace();
+                Timber.e(e, "Failed to validate account type %s", accountType);
```
Comment (davigonz) [inline owncloudApp/src/main/java/com/owncloud/android/authentication/AccountAuthenticator.java]: We are duplicating code here, please create a method with:

```
try {
    validateAccountType(accountType);
} catch (AuthenticatorException e) {
    Timber.e(e, "Failed to validate account type %s", accountType);
    return e.getFailureBundle();
}
```

Comment (abelgardep) [reply owncloudApp/src/main/java/com/owncloud/android/authentication/AccountAuthenticator.java]: I think that it is out of this pr scope which is to stop using printStacktrace.

BTW, we are going to refactor this class in the short term future with the new arch. So we can face this kind of problems there. Can't we?

Comment (davigonz) [reply owncloudApp/src/main/java/com/owncloud/android/authentication/AccountAuthenticator.java]: I try to use the boy scout rule (leave the code better than you found it) if the changes are tiny. I thought this was a good example but it looks good to me.

**Comment by davigonz** [comment]
@jesmrec code approved, ready to test

**Comment by jesmrec** [comment]
No much to test here, beyond the basic checks. Timber is making the magic.

Approved.

**Comment by hannesa2** [comment]
@davigonz 
Only for the records:
You could get rid of such pointless commits (which are in the history just noise) by using `--fixup`

Instead of 
`git commit -m "Rebase with master"`

perform a 
```
git commit --fixup 94ed4f6edd3
git rebase -i master --autosquash
```

**Comment by hannesa2** [comment]
Sorry, I mean this
image

**Comment by davigonz** [comment]
Thanks for the tip @hannesa2 , I'll give it a try in coming rebases with master

**Pull request merged**